### PR TITLE
[WIP][DNM] Spacesuit equip times

### PR DESCRIPTION
**Pull request by Ty-Omaha**
# **What does this PR do:**
This adds equip times for all space suits (including hardsuits) and the framework to have an equip time for anything else. You will have to remain uninterrupted while putting on most space suits. Certain advanced suits are still able to be equipped instantly. Certain suits are quicker than others.
Outfit equipping (admin select equipment, ERT spawning in) overrides any time set and will be equipped instantly.

The **slow** time is 6 seconds.
The **medium** time is 4 seconds
The **fast** time is 2 seconds.

### **Instant:**
Elite Syndicate
Shielded Syndicate 
Syndicate _Strike_ Team
Wizard
Deathsquad
All Plasmaman suits

### **Fast:**
Captain
Chief engineer
ERT
Security
Regular syndicate suits including the softsuits

### **Medium:**
Paramedic
Medical

### **Slow:**
Regular EVA
Engineering, mining, and anything else not included is **slow**

brtgopf

**Code review appreciated** (especially with the 5 different equip procs calling each other)

**Open to suggestions balance wise too**
🆑 
balance: Spacesuit equip times (including hardsuits). 
You will now have to stand still uninterrupted for a set period of time while equipping most spacesuits. 
Certain suits are faster than others.
For the full list see PR # 10649 on the github
/ 🆑

### CODE REVIEW

**Comment by KasparoVy** [comment]
Fuck yes

**Comment by Birdtalon** [comment]
Why though?

**Comment by Mitchs98** [comment]
I really don't see how this adds more depth or makes things more interesting at all really. Just adds a hurdle for scenarios similar to the described above. Also prevents crew from quickly evacing when needed into EVA in times a few seconds most definitely matter. This was never considered nor discussed in any external channels to a large capacity that I'm aware of, so just why?

More depth and making it interesting isn't a good reason without good proper examples as to well...why.

**Comment by Ty-Omaha** [comment]
I thought my example and having different tiers was good depth. Makes people think twice instead of just running around willy nilly. Opinions are subjective so not much more I can do there.

**Comment by Mitchs98** [comment]
...I meant evac from the station in an emergency. Not actual evac. And that's precisely what they meant as well.

**Comment by Ty-Omaha** [comment]
## What are people's opinions if this was inversed? (delay to unequip but equip)

## Also what are people's opinions if this was to both equip and unequip?


(bold text so it doesnt get buried)

**Comment by DesolateG** [comment]
This just feels like nerfing just to nerf with no valid balance reason. I don't see any interesting or adding depth about this, only an annoyance.

**Comment by Ty-Omaha** [comment]
>...I meant evac from the station in an emergency. Not actual evac.

@Mitchs98 Ah, that clears things up.

>And that's precisely what they meant as well.

That is not true though, the "literal end of a round" is when the shuttle docks at CC/Syndi and the score screen appears.

**Comment by Mitchs98** [comment]
To iterate a point made on the discord;

There's a delay to put on not take off. People misclick all the time. You're in EVA, possibly in unstable atmos conditions. Misclick. Instantly take off suit. Can normally put it back on before you get bitchslapped by space wind, but now you can't and in a lot of situations you're dead.

Same thing applies to combat in space and etc.

**Comment by Mitchs98** [comment]
Ty. They meant the literal end of a players round. AKA getting megafucked by whatever current threat there is because they couldn't EVA in time to escape.

If it was inversed it'd be a different story. It'd be a good change to help people that already misclick. Still not really all for it being on both, for reasons already stated. It'd be best to have it on both if you're REALLY pushing for it to be an equip delay, but otherwise I'd just prefer a removal delay.

**Comment by Allfd** [comment]
@Mitchs98 You have a timeline problem.  They did not mean the literal end to a players round.  They were removed months after we move to LINDA.  

Even if they were the cause, this would still be considered.  As we are looking at ways to make some of the danger in a round, not come from other players.

**Comment by Quantum-M** [comment]
Why not also add a delay on suit removal? That way they can't accidentally remove their suit.

**Comment by davidchan13** [comment]
I'd support delays on both ends (equipping and removing)

I feel the Cap and CE suit should not be faster than ERT suits, and that medical should be in the fast or medium category since entire point is mobility and rapid response.

**Comment by Rurik123** [comment]
I for one like this idea quite a lot. Delay for both equip and unequipping would be neat. 

I'm surprised theres so much discussion on this since it changes absolutely nothing balance wise. If anyone can bring up a secnario that refutes that, I'd be happy to change my mind there.

As for why, I'd argue for immersion, as depth doesn't really fit there. Or simply cause why not.

**Comment by Dave-TH** [comment]
I like this change a lot, nerfs the cheese of holding a spacesuit in your hand to avoid the slowdown.
If there is a delay to put it on, there should also be a delay to take it off to help mitigate misclicks (ie; accidentally ejecting yourself from your suit in space and being unable to put it back on in time).

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 1]
```diff
@@ -187,11 +187,18 @@
 
 //This is an UNSAFE proc. Use mob_can_equip() before calling this one! Or rather use equip_to_slot_if_possible() or advanced_equip_to_slot_if_possible()
 //set redraw_mob to 0 if you don't wish the hud to be updated - if you're doing it manually in your own proc.
-/mob/living/carbon/human/equip_to_slot(obj/item/W as obj, slot, redraw_mob = 1)
+/mob/living/carbon/human/equip_to_slot(obj/item/W as obj, slot, redraw_mob = 1, equip_speed_override = FALSE)
 	if(!slot) return
 	if(!istype(W)) return
 	if(!has_organ_for_slot(slot)) return
 
+	if(W.equip_speed && !equip_speed_override) //check to see if we want a delay and check to see we don't want an override for uses such as admin equipping (cmd_admin_dress).
```
Comment (Dave-TH) [inline code/modules/mob/living/carbon/human/inventory.dm]: Might be wise to give text feedback at the start of the bar as well.
ie; `to_chat(src, "<span class='notice'>You start to struggle into the [W].</span>")`

**Comment by variableundefined** [comment]
Fox and I don't believe this is a good change at all and we do not need to inconvenience using hardsuit more. As a result the PR will be closed.

**Comment by Ty-Omaha** [comment]
**In progress:**
Captain and CE suit to Fast
Medical to medium
Unified unequip time of 2 seconds - except for Instant equip suits

**Comment by Ty-Omaha** [comment]
Need to figure out the unequip logic

Before check_speed was inverted but it was being funky.

Now have to get check_speed to true when clicking it from the inventory

**Comment by Ralta** [comment]
Can it add a statement in local chat about the user entering/exiting hardsuits? 

E.g. 
"Jane Doe starts to remove the Engineering Hardsuit."
"John Smith starts to equip the Security Hardsuit."

Bit like entering/exiting mechs - feel like it'd be useful for nearby users to be able to see when someone is trying to put a suit on / take it off, either to interrupt them or to avoid interrupting them.

**Comment by datlo** [comment]
new meta if this gets merged : use atmospherics firesuit + coffee to heat the body instead of a hardsuit

if this gets merged (and I'm not sure I support it myself), this should be added to all "big" protective suit for consistency, like firesuits, riot/swat armor, bombsuits, probably with a fast/med speed instead of the long one

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -728,7 +728,7 @@
 	what.add_fingerprint(src)
 	if(do_mob(src, who, what.strip_delay))
 		if(what && what == who.get_item_by_slot(where) && Adjacent(who))
-			who.unEquip(what)
+			who.unEquip(what, TRUE) //TRUE to override the unequip delay on certain items
```
Comment (Ty-Omaha) [inline code/modules/mob/living/living.dm]: this should actually be false? unless the thing is inversed again

**Comment by Ty-Omaha** [comment]
I need help finding where inventory unequip is located

**Comment by KasparoVy** [comment]
Inventory? Do you mean the unequip proc on space suit objects?

**Comment by Ty-Omaha** [comment]
@KasparoVy yeah, for when you unequip from inventory, not.. let's say strip someone

File: code/modules/mob/living/carbon/human/inventory.dm (hunk 1 above)
Comment (MrMagolor) [reply code/modules/mob/living/carbon/human/inventory.dm]: Would also make sense if the text corresponded to the speed for that suit type- i.e., a mining hardsuit would use the struggle example above, but a syndicate elite suit(instant) would possibly say, "You slip into the syndicate elite hardsuit."

**Comment by variableundefined** [comment]
I'll be closing this PR for the same reasons stated above - We do not believe these changes to be good for hardsuit.

**Pull request closed without merging**